tests/basic/pthread_detach/README.md

```diff
@@ -0,0 +1,181 @@
+# Preliminaries
+## Thread identities
+A thread has 3 identities:
+- Linux process context - `task_struct`
+- LKL arch specific process context - `thread_info`
+- LKL host thread context, implemented in SGX-LKL by `lthread`s.
+
+For most of a thread's lifetime these 3 identities should have a 1:1 mapping between them.
+One instance where this doesn't hold is on the return path of the clone() system call. (This is covered in more detail later in the `pthread_create` section.)
```

Comment:
What is the purpose of introducing 1 instance here? To me, its a bit confusing and could be covered later.

My impression is that you are trying to establish basic intutions. I think you would be best served by that. Something like "in general, you can think of this as... BLAH" and then "we'll detail some exceptions".

And perhaps something to the effect of "these rules arent' really true but they can ease you into understanding the more complicated picture better"